include/AudioEngineProfilerProbe.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * AudioEngineProfilerProbe.h - RAII-style probe for AudioEngineProfiler
+ *
+ * Copyright (c) 2023 Martin Pavelek <lmms@he29.net>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef LMMS_AUDIO_ENGINE_PROFILER_PROBE_H
+#define LMMS_AUDIO_ENGINE_PROFILER_PROBE_H
+
+#include "AudioEngineProfiler.h"
+
+namespace lmms
+{
+
+class AudioEngineProfilerProbe
+{
+public:
+	AudioEngineProfilerProbe(AudioEngineProfiler& profiler, AudioEngineProfiler::DetailType type);
+	~AudioEngineProfilerProbe();
+
+private:
+	AudioEngineProfiler &m_profiler;
+	const AudioEngineProfiler::DetailType m_type;
```

Comment:
I did not realize the reference would affect copying etc., I just tend to avoid pointers when they are not necessary. I can change it to a pointer if you think it's a better practice.

The constant `m_type` is intentional: since the timer of a given type is started in the constructor, it should not be possible to modify the type afterwards -- to ensure that the destructor is always called on the same timer. The `const` seemed like the right way how to formalize this intention.

Reply:
If you're not changing the const, there's no point changing the reference, so it's probably best left as is, since references are indeed preferable to pointers in general where possible.

The issue is that assigning to an object requires overwriting its member variables with new values, and references and const objects cannot be written. Even if member variables should stay constant between assignments to an object, they need to be mutable to support any eventual assignment. Another impacted scenario is move construction - even if assignment is not supported, this still typically requires clearing out the source object to prevent its destructor doing what is now the responsibility of the new object.

As mentioned in the comment above, this is fine in this class as it is specifically designed to be immovable and unassignable, but usually ought to be avoided even if it would better express invariants, as it significantly impacts the usability of the containing class.